src/uharfbuzz/_harfbuzz.pyx

```diff
@@ -619,21 +618,14 @@ cdef class FontFuncs:
         self._nominal_glyph_func = func
 
 
-cdef const char ** to_cstring_array(list_str):
-    cdef const char **ret = <const char **>malloc(len(list_str) * sizeof(char *))
-    for i in range(len(list_str)):
-        ret[i] = PyUnicode_AsUTF8(list_str[i])
-    return ret
-
-
 def shape(font: Font, buffer: Buffer,
         features: Dict[str,Union[int,bool,Sequence[Tuple[int,int,Union[int,bool]]]]] = None,
         shapers: List[str] = None) -> None:
     cdef unsigned int size
     cdef hb_feature_t* hb_features
     cdef bytes packed
     cdef hb_feature_t feat
-    cdef const char **c_shapers
+    cdef const char* c_shapers[10]
```

Comment:
Is this WIP or is this a revision on your proposed fix from https://github.com/harfbuzz/uharfbuzz/pull/84#issuecomment-885965144

Reply:
Sorry, I only now saw the commit msg! If you think 9 is a good number, let's go with this!

Reply:
It should be plenty, HarfBuzz currently has 6 shapers and the number is expected to go down not up.